tests/Adapter/Http/FileHttpClient.php

```diff
@@ -42,17 +42,19 @@ public function request($method, $path, $body = null, array $options = [])
     public function asyncRequest($method, $path, $body = null, array $options = [])
     {
         $filePath = $this->fileResolver->getFilePath($method, $path, $body, $options);
-        return new Promise(
-            function () use ($filePath) {
+        /** @var PromiseInterface $promise */
+        $promise = new Promise(
+            function () use (&$promise, $filePath) {
                 if (!file_exists($filePath)) {
                     throw new \RuntimeException(sprintf(
                         'The HTTP fixture file "%s" do not exists',
                         $filePath
                     ));
                 }
 
-                return file_get_contents($filePath);
+                $promise->resolve(file_get_contents($filePath));
```

Comment:
Actually, that was what the `return` was doing.. not sure why you changed that :)

Reply:
Because it did not work. The promise failed with an error like "**Invoking the wait callback did not resolve the promise**".
Maybe a spec is needed for this as well.